src/mixins/amalgamation-mixin.ts

```diff
@@ -52,6 +55,45 @@ export default class AmalgamationMixin extends Vue {
   //   }
   // ]
 
+  /**
+   * Re-fetch the draft amalgamating businesses information and set in the store.
+   * Need to do that because the businesses might have changed from last draft save.
+   */
+  async refetchBusinessesInfo (): Promise<void> {
+    async function fetchAndMapBusinesses (item: any): Promise<AmalgamatingBusinessIF> {
+      let business = null
+      const data = await Promise.all([
+        LegalServices.fetchBusinessInfo(item.identifier),
+        AuthServices.fetchAuthInfo(item.identifier),
+        LegalServices.fetchAddresses(item.identifier)
+      ]).catch((error) => {
+        return error
+      })
+      if (data.length === 3) {
+        business = data[0].data?.business
+        business.businessContact = data[1].contacts[0]
+        business.officeAddress = data[2]
+      }
+      if (business) {
+        const tingBusiness = {
+          type: 'lear',
+          role: AmlRoles.AMALGAMATING,
+          identifier: business.identifier,
+          name: business.legalName,
+          email: business.businessContact.email,
+          legalType: business.legalType,
+          address: business.officeAddress.registeredOffice.mailingAddress,
+          goodStanding: business.goodStanding
+        } as AmalgamatingBusinessIF
+        return tingBusiness
+      }
+    }
+
+    this.setAmalgamatingBusinesses(
+      await Promise.all(this.getAmalgamatingBusinesses.map(fetchAndMapBusinesses))
```

Comment:
Running the mapping in parallel for better performance.

Reply:
🤯

Reply:
OK, the only thing is I've been cleaning this code up. Maybe I should merge my changes in again...

See https://github.com/bcgov/business-create-ui/pull/593 and tell me what you think.

Also, I think we should try to have a single instance of the "fetch" code.

Reply:
Sure thing Sev! I can rebase after you merge.

I'll try to simplify and have the code only in one place.

Reply:
Shall we resolve this conversation now?

Reply:
Yup, this should be done now.